backend/file-system/src/main/java/com/htec/filesystem/service/ShelfService.java

```diff
@@ -57,32 +61,34 @@ public void createShelf(CreateShelfRequestModel createShelfRequestModel, Long us
     }
 
     @Transactional
-    public void softDeleteShelf(AuthUser user, Long shelfId) {
+    public void softDeleteShelf(AuthUser user, List<Long> shelfIds) {
```

Comment:
This method could be optimized

Reply:
Try getting first all schelfs, then all folders and then all files, and not one by one, after that go trough all of those set flag and save all of them

Reply:
Should i resolve this in sql?